allennlp/training/metrics/conll_scores.py

```diff
@@ -0,0 +1,155 @@
+from typing import Tuple
+from collections import Counter
+import numpy as np
+from sklearn.utils.linear_assignment_ import linear_assignment
+
+from overrides import overrides
+
+from allennlp.training.metrics.metric import Metric
+
+@Metric.register("conll_scores")
+class ConllScores(Metric):
+    def __init__(self) -> None:
+        self.reset()
+
+    @overrides
+    def __call__(self, top_spans, antecedent_indices, predicted_antecedents, metadata_list):
+        top_spans, antecedent_indices, predicted_antecedents = self.unwrap_to_tensors(top_spans,
+                                                                                      antecedent_indices,
+                                                                                      predicted_antecedents)
+        for i, metadata in enumerate(metadata_list):
+            gold_clusters, mention_to_gold = self.get_gold_clusters(metadata["clusters"])
+            predicted_clusters, mention_to_predicted = self.get_predicted_clusters(top_spans[i],
+                                                                                   antecedent_indices,
+                                                                                   predicted_antecedents[i])
+            for scorer in self.scorers:
+                scorer.update(predicted_clusters, gold_clusters, mention_to_predicted, mention_to_gold)
+
+    @overrides
+    def get_metric(self, reset: bool = False) -> Tuple[float, float, float]:
+        metrics = (lambda e: e.get_precision(), lambda e: e.get_recall(), lambda e: e.get_f1())
+        precision, recall, f1_score = tuple(sum(m(e) for e in self.scorers) / len(self.scorers) for m in metrics)
+        if reset:
+            self.reset()
+        return precision, recall, f1_score
+
+    @overrides
+    def reset(self):
+        self.scorers = [Scorer(m) for m in (Scorer.muc, Scorer.b_cubed, Scorer.ceafe)]
```

Comment:
I'm surprised pylint isn't complaining about `self.scorers` being defined outside of `__init__`.

Reply:
Fixed, since I'm assuming the implication is that it's better to duplicate the call in `reset` and `__init__`.

Reply:
I wasn't trying to be passive-aggressive with that, I was just surprised (and still am) that pylint wasn't doing what I expected there.  I don't have a strong preference in this case.  Had pylint complained, I probably would have just done `self.scorers = []; self.reset()` in the constructor, to avoid the duplication, but what you have is fine.